### PR TITLE
Skip @Ignore tests before send it to ftl and reapply to local JUnit Xml

### LINKED ISSUE #852: Skip `@Ignore` tests before send them to FTL and add to final JUnitResult
> 
If I understand the request in https://github.com/Flank/flank/issues/629, the user wants `@Ignore` tests represented in the XML. Maybe we could do the following:

- Filter all `@Ignore` / skipped / etc. tests before sending them to FTL
- Keep track in Flank of all the Ignored tests
- FTL runs the matrix, and at the end Flank adds in the `@Ignore` tests to the final merged XML report

_Originally posted by @bootstraponline in https://github.com/Flank/flank/issues/818#issuecomment-643661984_

### PR DESCRIPTION
**Pull request by adamfilipow92**
Fixes #852 

.

Ignored test should't shows on FTL

## Checklist

- [X] Unit tested
- [X] release_notes.md updated
- [X] Add ignored tests to test app

### CODE REVIEW

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt [hunk 1]
```diff
@@ -48,13 +48,16 @@ private fun InstrumentationTestContext.downloadApks(): InstrumentationTestContex
...
-        args = args,
-        forcedShardCount = args.numUniformShards
-    ).filter { it.isNotEmpty() }
-)
+): InstrumentationTestContext = ArgsHelper.calculateShards(
+    filteredTests = getFlankTestMethods(testFilter),
+    args = args,
+    forcedShardCount = args.numUniformShards
+).run {
+    return@run copy(
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:56]: Unnecessary `return@run`

File: test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt [hunk 2]
```diff
@@ -48,13 +48,16 @@ private fun InstrumentationTestContext.downloadApks(): InstrumentationTestContex
...
+        shards = this.shardChunks.filter { it.isNotEmpty() },
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt:57]: Unnecessary `this`

File: test_runner/src/main/kotlin/ftl/reports/util/ReportManager.kt [hunk 3]
```diff
@@ -114,14 +123,27 @@ object ReportManager {
                 HtmlErrorReport
             ).map { it.run(matrices, testSuite, printToStdout = false, args = args) }
         }
-        JUnitReport.run(matrices, testSuite, printToStdout = false, args = args)
+        JUnitReport.run(matrices, testSuite.apply {
```
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/reports/util/ReportManager.kt:126]:
```
        JUnitReport.run(matrices, testSuite?.apply {
            if (ignoredTestCases.isNotEmpty()) {
                testsuites?.add(ignoredTestCases.toJunitTestsResults())
            }
        }, printToStdout = false, args = args)
```

File: release_notes.md [hunk 4]
```diff
@@ -1,6 +1,9 @@
 ## next (unreleased)
 
+- [#853](https://github.com/Flank/flank/pull/853) Skip @Ignore tests before send it to ftl and reapply to local JUnit Xml. ([piotradamczyk5](https://github.com/piotradamczyk5), [adamfilipow92](https://github.com/adamfilipow92))
```
Comment (bootstraponline) [inline release_notes.md:3]:
```suggestion
- [#853](https://github.com/Flank/flank/pull/853) Store @Ignore tests in the JUnit XML without sending ignored tests to FTL. ([piotradamczyk5](https://github.com/piotradamczyk5), [adamfilipow92](https://github.com/adamfilipow92))
```

**Pull request merged**